### ISSUE TITLE
Steam Version: Cannot Boot. SharpDX Exception

### ISSUE THREAD

**Issue by BarryCarlyon** [Regalis11/Barotrauma#1698]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
On hitting play, I'm presented with two .Net Framework Error logs one after the other

**Steps To Reproduce**
Boot the game

**Version**
Windows 10 (v1903)
v0.9.0.6  according to Version.txt

**Additional information**

I was wondering if it was a Display Dimension issue. As I'm on an ultrawide.

I tried 

```
  <graphicsmode
    width="1920"
    height="1080"
    vsync="true"
    displaymode="BorderlessWindowed" />
```

in both `config.xml` and `config_player.xml` which did not help (same error)

Copy/paste of the two errors follows:

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
SharpDX.SharpDXException: HRESULT: [0x80070057], Module: [General], ApiCode: [E_INVALIDARG/Invalid Arguments], Message: The parameter is incorrect.

   at SharpDX.Result.CheckError()
   at SharpDX.DXGI.SwapChain.ResizeBuffers(Int32 bufferCount, Int32 width, Int32 height, Format newFormat, SwapChainFlags swapChainFlags)
   at Microsoft.Xna.Framework.Graphics.GraphicsDevice.CreateSizeDependentResources()
   at Microsoft.Xna.Framework.Graphics.GraphicsDevice.Reset()
   at MonoGame.Framework.WinFormsGameWindow.UpdateBackBufferSize()
   at MonoGame.Framework.WinFormsGameWindow.OnResize(Object sender, EventArgs eventArgs)
   at System.Windows.Forms.Control.OnResize(EventArgs e)
   at System.Windows.Forms.Form.OnResize(EventArgs e)
   at System.Windows.Forms.Control.OnSizeChanged(EventArgs e)
   at System.Windows.Forms.Control.UpdateBounds(Int32 x, Int32 y, Int32 width, Int32 height, Int32 clientWidth, Int32 clientHeight)
   at System.Windows.Forms.Control.UpdateBounds()
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Form.WndProc(Message& m)
   at Microsoft.Xna.Framework.Windows.WinFormsGameForm.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
Barotrauma
    Assembly Version: 0.9.0.6
    Win32 Version: 0.9.0.6
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/Barotrauma.exe
----------------------------------------
MonoGame.Framework
    Assembly Version: 3.7.1.189
    Win32 Version: 3.7.1.189
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/MonoGame.Framework.DLL
----------------------------------------
Facepunch.Steamworks
    Assembly Version: 1.2.0.0
    Win32 Version: 1.2.0.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/Facepunch.Steamworks.DLL
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
FarseerPhysics MonoGame
    Assembly Version: 3.5.0.37723
    Win32 Version: 3.5.0.*
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/FarseerPhysics%20MonoGame.DLL
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
SharpDX
    Assembly Version: 4.2.0.0
    Win32 Version: 4.2.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/SharpDX.DLL
----------------------------------------
SharpDX.XAudio2
    Assembly Version: 4.2.0.0
    Win32 Version: 4.2.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/SharpDX.XAudio2.DLL
----------------------------------------
System.Xml.Linq
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml.Linq/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.Linq.dll
----------------------------------------
SharpDX.DXGI
    Assembly Version: 4.2.0.0
    Win32 Version: 4.2.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/SharpDX.DXGI.DLL
----------------------------------------
SharpDX.Direct3D11
    Assembly Version: 4.2.0.0
    Win32 Version: 4.2.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/SharpDX.Direct3D11.DLL
----------------------------------------
Hyper.ComponentModel
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/Hyper.ComponentModel.DLL
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
```

and

```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
SharpDX.SharpDXException: HRESULT: [0x80070057], Module: [General], ApiCode: [E_INVALIDARG/Invalid Arguments], Message: The parameter is incorrect.

   at SharpDX.Result.CheckError()
   at SharpDX.DXGI.SwapChain.ResizeBuffers(Int32 bufferCount, Int32 width, Int32 height, Format newFormat, SwapChainFlags swapChainFlags)
   at Microsoft.Xna.Framework.Graphics.GraphicsDevice.CreateSizeDependentResources()
   at Microsoft.Xna.Framework.Graphics.GraphicsDevice.Reset()
   at MonoGame.Framework.WinFormsGameWindow.UpdateBackBufferSize()
   at MonoGame.Framework.WinFormsGameWindow.OnResize(Object sender, EventArgs eventArgs)
   at System.Windows.Forms.Control.OnResize(EventArgs e)
   at System.Windows.Forms.Form.OnResize(EventArgs e)
   at System.Windows.Forms.Control.OnSizeChanged(EventArgs e)
   at System.Windows.Forms.Control.UpdateBounds(Int32 x, Int32 y, Int32 width, Int32 height, Int32 clientWidth, Int32 clientHeight)
   at System.Windows.Forms.Control.UpdateBounds()
   at System.Windows.Forms.Control.WmWindowPosChanged(Message& m)
   at System.Windows.Forms.Control.WndProc(Message& m)
   at System.Windows.Forms.Form.WndProc(Message& m)
   at Microsoft.Xna.Framework.Windows.WinFormsGameForm.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
Barotrauma
    Assembly Version: 0.9.0.6
    Win32 Version: 0.9.0.6
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/Barotrauma.exe
----------------------------------------
MonoGame.Framework
    Assembly Version: 3.7.1.189
    Win32 Version: 3.7.1.189
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/MonoGame.Framework.DLL
----------------------------------------
Facepunch.Steamworks
    Assembly Version: 1.2.0.0
    Win32 Version: 1.2.0.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/Facepunch.Steamworks.DLL
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
FarseerPhysics MonoGame
    Assembly Version: 3.5.0.37723
    Win32 Version: 3.5.0.*
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/FarseerPhysics%20MonoGame.DLL
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
SharpDX
    Assembly Version: 4.2.0.0
    Win32 Version: 4.2.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/SharpDX.DLL
----------------------------------------
SharpDX.XAudio2
    Assembly Version: 4.2.0.0
    Win32 Version: 4.2.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/SharpDX.XAudio2.DLL
----------------------------------------
System.Xml.Linq
    Assembly Version: 4.0.0.0
    Win32 Version: 4.8.3752.0 built by: NET48REL1
    CodeBase: file:///C:/WINDOWS/Microsoft.Net/assembly/GAC_MSIL/System.Xml.Linq/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.Linq.dll
----------------------------------------
SharpDX.DXGI
    Assembly Version: 4.2.0.0
    Win32 Version: 4.2.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/SharpDX.DXGI.DLL
----------------------------------------
SharpDX.Direct3D11
    Assembly Version: 4.2.0.0
    Win32 Version: 4.2.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/SharpDX.Direct3D11.DLL
----------------------------------------
Hyper.ComponentModel
    Assembly Version: 1.0.0.0
    Win32 Version: 1.0.0.0
    CodeBase: file:///E:/Steam/steamapps/common/Barotrauma/Hyper.ComponentModel.DLL
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
```

**Comment by BarryCarlyon** [comment]
Update on issue

Game Version: v0.9.0.7 

On launch, silent closes/crashes with no error

**Comment by itchyOwl** [comment]
Sorry for the late response.

You wrote that you are using an "ultrawide". What is the exact display resolution you are trying to use? Did you try to switch it in-game? Is the game crashing everytime you try to launch it? Did you try to reinstall the game and update the display drivers? Which GPU and driver are you on?

**Comment by BarryCarlyon** [comment]
I cannot get into game, to change any settings

On v0.9.0.6 hitting launch on steam. It would get a dialog popup. To which I copied/pasted out the errors and pasted them in OP
Since v0.9.0.7 upwards, it just silents crashes and closes.

My screen resolution on the primary display is: 5431x1527
I'm running the latest Nvidia drivers: v 431.36 (07/09/2019)

As mentioned in OP I modified the configXML's to try to launch at a different resolution and in BorderlessWindowed, or even Windowed and got the same crash dialog (and/or silent crash)

  <graphicsmode
    width="1920"
    height="1080"
    vsync="true"
    displaymode="BorderlessWindowed" />

I have verified the files which then reset the changed XML's and it still silent crashes.

In steam I just uninstalled and then reinstalled this second.

Silent Crashes.

**Comment by BarryCarlyon** [comment]
Confirming as still broken under v0.9.2.1 

I got todays patch. Tried to run.
Verfied files. Tried to run.
Reinstalled. Tried to run.

Steam goes "starting". I get no window/nothing Steam just goes back to "not running"

**Comment by BarryCarlyon** [comment]
Still broken on v0.9.3.3

**Comment by BarryCarlyon** [comment]
Seems fixed in v0.9.5.1 

I got to the main menu!
